package/kernel-osmc/initramfs-src/init

```diff
@@ -83,8 +83,37 @@ for option in $(/bin/busybox cat /proc/cmdline); do
 	    ;;
 	  root=*)
 		OPTION_ROOT=$(echo "${option#*=}" | /bin/busybox sed 's/"//g')
+		ROOT_STR=$OPTION_ROOT
 		PARTUUID=$(echo "$OPTION_ROOT" | /bin/busybox grep "^PARTUUID=" | /bin/busybox sed 's/^PARTUUID=//g')
-		if [ -n "$PARTUUID" ]; then OPTION_ROOT=$(/bin/busybox readlink -f /dev/disk/by-partuuid/${PARTUUID}); fi
+		UUID=$(echo "$OPTION_ROOT" | /bin/busybox grep "^UUID=" | /bin/busybox sed 's/^UUID=//g')
+		if ! [ -z "$UUID$PARTUUID" ]; then
+			# it takes a while for USB partitions to show up
+			count=12
```

Comment:
Is there a specific reason 12 is a good number? I think even 30 could be acceptable for slow spinning  drives.

Reply:
In testing USB sticks and some ancient 2.5" HDDs in caddies, 8-9 secs was enough.  30 secs is a long time to wait with a blank screen which would happen if, eg you booted from SD card with the root device not connected (or powered) or the UUID was wrong.